mkdocs_blogging_plugin/plugin.py

```diff
@@ -114,6 +124,27 @@ def read_in_config(self, global_config):
         # Read in configs of all categories
         self.categories["global"] = BloggingConfig(self.config)
         categories = self.config.get("categories")
+        docs_dir = Path(global_config.get("docs_dir"))
+        if self.use_root_dirs:
+            root_dirs = [d for d in docs_dir.iterdir() if d.is_dir()]
+            for d in root_dirs:
+                config = BloggingConfig(self.config)
+                config.dirs = [d.name]
+                self.categories[d.name] = config
+        if self.config_filename:
+            config_files = [f for f in docs_dir.rglob(self.config_filename)]
+            for f in config_files:
+                with open(f, "r") as stream:
+                    config = yaml.safe_load(stream)
+                this_categories = config.get("categories", None)
```

Comment:
`categories` is a list of categories (each of them is a dict) in our settings. But it seems that you treat this entry as the config of a single category (dict).